From b405725794ac71642463a66bd87ab1430a0f90df Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 15 Nov 2011 11:54:14 +0100
Subject: [PATCH] 40network/ifup: add "brd +" to "ip addr add"

Unlike "ifconfig", the "ip" command does not set a broadcast address
unless explicitly requested.

add "brd +" to make "ip" calculate the broadcast address on the fly

backport of commit d1b7efd57874e61d4ad3792fea56cf0979d9dc6b

https://bugzilla.redhat.com/show_bug.cgi?id=75258
---
 modules.d/40network/ifup | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/40network/ifup b/modules.d/40network/ifup
index 9eab3b4..4c73da8 100755
--- a/modules.d/40network/ifup
+++ b/modules.d/40network/ifup
@@ -66,7 +66,7 @@ do_static() {
 	# do not flush addr for ipv6
 	#strstr $ip '*:*:*' || \
 	#    ip addr flush dev $netif
-	ip addr add $ip/$mask dev $netif
+	ip addr add $ip/$mask brd + dev $netif
     } 
     > /tmp/net.$netif.up
 
